### PR TITLE
Authenticate users and respond with provide access_token token

### PR DESCRIPTION
**Pull request by hurrycane**
Making authentication work more like an API authentication will.

I'm planning to move all devise stuff to /internal and leave them there for now. Also user roles will be applied so who will sign up using internal will not be granted access to admin.

### CODE REVIEW

**Comment by vtemian** [comment]
LGTM

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -47,4 +47,35 @@ def cors_preflight_check
...
+
+    if user && Devise.secure_compare(user.access_token, auth_token)
+      sign_in user, store: false
+    else
+      authentication_error
+    end
+  end
+
+  def authentication_error
+    render json: { error: 'unauthorized' }, status: 401
```
Comment (vtemian) [inline app/controllers/application_controller.rb]: Yey! We have good error codes!

**Comment by robertdolca** [comment]
Can we remove the automatic admin access? That should be done manually. That way we do not need to have two sign up routes.

**Comment by hurrycane** [comment]
Read my initial description @robertdolca. I will remove automatic admin access once I get roles in.

Also this is a sign in route. For starters we need two auth routes.

One that is HTTP + json only and sits in /v1/sign_in
Another one that is HTML and sits in /internal/users/sign_in.

We're using the 2nd one to log in into admin.

**Comment by robertdolca** [comment]
> Read my initial description @robertdolca. I will remove automatic admin access once I get roles in.
I misread the initial comment. Everything is ok.

**Pull request merged**